runtime/lua/vim/_defaults.lua

```diff
@@ -23,7 +23,19 @@ do
 
   vim.api.nvim_create_user_command('EditQuery', function(cmd)
     vim.treesitter.query.edit(cmd.fargs[1])
-  end, { desc = 'Edit treesitter query', nargs = '?' })
+  end, {
+    desc = 'Edit treesitter query',
+    nargs = '?',
+    complete = function()
```

Comment:
Let's put this logic in `vim.treesitter.query._complete()`. Need to keep defaults.lua as simple as possible.